server/core/hot_region_storage_test.go

```diff
@@ -177,17 +227,112 @@ func (t *testHotRegionStorage) TestHotRegionDelete(c *C) {
 	packHotRegionInfo.historyHotReads = historyHotRegions
 	store.pullHotRegionInfo()
 	store.flush()
-	store.delete()
+	store.delete(defaultRemainDay)
 	iter := store.NewIterator(HotRegionTypes,
 		deleteDate.UnixNano()/int64(time.Millisecond),
 		time.Now().UnixNano()/int64(time.Millisecond))
 	num := 0
 	for next, err := iter.Next(); next != nil && err == nil; next, err = iter.Next() {
 		num++
-		c.Assert(reflect.DeepEqual(next, &historyHotRegions[defaultReaminDay-num]), IsTrue)
+		c.Assert(reflect.DeepEqual(next, &historyHotRegions[defaultRemainDay-num]), IsTrue)
 	}
 }
 
+func (t *testHotRegionStorage) checkConfigChangeResult(now time.Time,
+	store *HotRegionStorage,
+	historyHotRegions []HistoryHotRegion,
+	expectedLength int,
+	c *C) {
+	iter := store.NewIterator([]string{ReadType.String()},
+		now.UnixNano()/int64(time.Millisecond),
+		now.Add(40*time.Second).UnixNano()/int64(time.Millisecond))
+	index := 0
+	for next, err := iter.Next(); next != nil && err == nil; next, err = iter.Next() {
+		c.Assert(err, IsNil)
+		c.Assert(reflect.DeepEqual(&historyHotRegions[index], next), IsTrue)
+		index++
+	}
+	c.Assert(index, Equals, expectedLength)
+}
+
+// Wait three times as previous interval to make sure ticker trigger,
+// Every trigger will reset new interval and pull hot regions.
+// hot regions: [1]
+// 100ms 0 days, record 0
+// 100ms 7 days, record 1
+// hot regions: [1,2]
+//    1s 7 days, record 2
+// 100ms 0 days, record 2
+// hot regions: [1,2,3]
+// 100ms 7 days, record 2
+// 100ms 7 days, record 3
+func (t *testHotRegionStorage) TestHotRegionConfigChange(c *C) {
+	packHotRegionInfo := &MockPackHotRegionInfo{isLeader: true}
+	// 100 ms 0 days
+	store, clean, err := newTestHotRegionStorage(100*time.Millisecond, 0, packHotRegionInfo)
+	c.Assert(err, IsNil)
+	defer clean()
+	now := time.Now()
+	historyHotRegions := []HistoryHotRegion{
+		{
+			UpdateTime: now.UnixNano() / int64(time.Millisecond),
+		},
+		{
+			UpdateTime: now.Add(10*time.Second).UnixNano() / int64(time.Millisecond),
+		},
+		{
+			UpdateTime: now.Add(20*time.Second).UnixNano() / int64(time.Millisecond),
+		},
+	}
+	packHotRegionInfo.SetHistoryHotReadRegions(historyHotRegions[:1])
+	// wait for 300ms, three times as previous interval
+	time.Sleep(300 * time.Millisecond)
+	// reserved days is 0, record 0 hot regions
+	t.checkConfigChangeResult(now, store, historyHotRegions, 0, c)
+
+	// 100ms 7 days
+	packHotRegionInfo.SetHotRegionsReservedDays(7)
+	// wait for 300ms
+	time.Sleep(300 * time.Millisecond)
+	// reserved days change to 7, record 1 hot regions
+	t.checkConfigChangeResult(now, store, historyHotRegions, 1, c)
```

Comment:
The test only test the mock.It cannot verify whether this configuration is really take effective.

Reply:
add unit test for config change